File: backend/src/InsuraTech.Application/HomePlans/DTOs/HomeQuotationDto.cs
namespace InsuraTech.Application.HomePlans.DTOs;

public sealed record HomeQuotationDto
{
    public required decimal                 PropertyValue       { get; init; }
    public required int                     ConstructionYear    { get; init; }
    public required int                     PropertyAge         { get; init; }
    public required int                     Stratum             { get; init; }
    public required int                     Occupants           { get; init; }
    public required string                  PropertyType        { get; init; }
    public required decimal                 BaseMonthlyPremium  { get; init; }
    public required IReadOnlyList<string>   SelectedCoverages   { get; init; }
    public required IReadOnlyList<string>   AppliedMultipliers  { get; init; }
    public required decimal                 FinalMonthlyPremium { get; init; }
}

File: backend_v2/src/main/java/com/insuratech/application/homeplans/dto/HomeQuotationDto.java
package com.insuratech.application.homeplans.dto;

import java.math.BigDecimal;
import java.util.List;

public record HomeQuotationDto(
    BigDecimal propertyValue,
    int constructionYear,
    int propertyAge,
    int stratum,
    int occupants,
    String propertyType,
    BigDecimal baseMonthlyPremium,
    List<String> selectedCoverages,
    List<String> appliedMultipliers,
    BigDecimal finalMonthlyPremium
) {}
